### DISCUSSION TITLE
I don't understand...

### DISCUSSION THREAD

**Discussion by artydev** [gnh1201/welsonjs#107] [General]
Hello,
I absolutely don't understand how all this work....
But  I am interested in it.
I have used HTA  many years ago.

Can you  give a step by step example ?

Regards

**Comment by gnh1201** [comment]
Thank you for leaving a discussion.

Simply run the "bootstrap.bat" file, and all processes will be completed.

If you want to create a customized GUI app (HTA-based), edit the files index.js, index.html, and style.css under the app folder.

**Comment by aolko** [comment]
more examples please?
How do i customize the window frame?
What's the bare minimum distribution for the app itself? How to package the app?

**Reply by gnh1201** [reply]
Hello. The WelsonJS framework allows you to do everything with HTML/CSS and JavaScript changes. While it is usually sufficient to distribute it as a zip file, [an example of creating an installer using Inno Setup](https://github.com/gnh1201/welsonjs/blob/master/setup.iss) is also included, which might be helpful for you. Since the minimum distribution only needs to include the JavaScript code, it is expected to be very small. As of version 0.2.7.14, including all image assets, the zip file is approximately 10MB in size.

If you have any questions, please feel free to ask through our public channels. Thank you.

**Reply by aolko** [reply]
post the minimal distribution file tree

**Reply by gnh1201** [reply]
The minimum distribution consists of only two files: `app.js` and `lib/pipe-ipc.js`.

- **app.js**: This is the main file of the framework.
- **lib/pipe-ipc.js**: This file handles all standard input and output operations, as well as character encoding.

**Reply by aolko** [reply]
alright, is there a similar way to bundle that in the single app executable, __not installer__?

**Reply by gnh1201** [reply]
> alright, is there a similar way to bundle that in the single app executable, **not installer**?

I have examples for EXE packaging, but I have not yet reflected them in the documentation. I will update it soon. Thank you.

**Reply by gnh1201** [reply]
There have been various attempts at EXE packaging for WelsonJS, and I've reviewed as much feedback as possible. This includes not only usability but also whether the process is relatively secure.

I have tentatively concluded that a launcher method, where a ZIP file is distributed separately (which I have decided to call the "WelsonJS Application Package") and then executed, would be effective. Therefore, I have begun distributing the [WelsonJS.Launcher](https://github.com/gnh1201/welsonjs/wiki/Awesome-binaries) artifact, which you may want to take note of.

You can download and run the release or upstream source code of this project directly from GitHub in ZIP file format.

**Comment by gnh1201** [comment]
As of now, we can use the [WelsonJS.Launcher](https://github.com/gnh1201/welsonjs/wiki/Awesome-binaries). You can run it by providing the project files (compressed in a ZIP file).

![Screenshot of WelsonJS.Launcher](https://github.com/user-attachments/assets/8bdfad9d-66e1-427b-b7db-4d1b2728e338)
